### ISSUE TITLE
Compatibility with User Themes shell estension

### ISSUE THREAD

**Issue by marmis85** [micheleg/dash-to-dock#241]
When I use your extension in expanded mode with a theme for the User Themes extension there is a small margin between the dash and the shell top bar
![shell](https://cloud.githubusercontent.com/assets/1887387/10560399/8f89c628-750a-11e5-873d-92f8e3e5b6b7.png)

I noticed that if I turn OFF/ON Dash to dock with gnome tweak tool, the margin disappears; this led me to think that it's an issue caused by Dash to dock loading before the User themes extension.
I don't know if there is a nice way to control the load order or even know if an extension has finished loading, but as a super ugly workaround, I renamed the enable() function to _enable() and wrote this enable() function:
```javascript
...
const GLib = imports.gi.GLib;
...

function enable() {
    GLib.timeout_add(1, 1, _enable)
}

function _enable() {
...
```
I also tried with longer timeouts but 1ms seems enough to do the trick for me

**Comment by micheleg** [comment]
Strange, I'm monitoring for theme changes so loading before the custom theme should not be a problem. With which theme and with which extension version are you experiencing the problem?

**Comment by marmis85** [comment]
I'm using the latest version available from the extensions.gnome.org website for gnome 3.18, its version 49 in the manifest. Actually I've been experiencing the issue since the big update to panel mode and settings (it was around the release of gnome 3.14 IIRC)
The theme I use is [elegance-colors](https://github.com/satya164/elegance-colors)
It could be a theme-specific issue as I tried installing a couple random shell themes and they work fine.
It's strange that my hack even works as 1ms shouldn't make any difference

**Comment by micheleg** [comment]
This bug should have incidentally been solved by 58164378d, so if you update to current master the problem should be gone.

Note to myself: I before commit 58164378d I wasn't repositioning the dock on theme change. However, even doing so the dock retrieves the wrong panel height, probably because some actual style is still not applied.

**Comment by marmis85** [comment]
I tried the latest master, and I confirm it works.
Thank you very much!
